syntax = "proto3";

service UserService {
  rpc GetUserById(GetUserRequest) returns (User);
  rpc CreateUser(CreateUserRequest) returns (User);
  rpc LoginUser(LoginRequest) returns (User);
  rpc CheckUsername(UsernameRequest) returns (UsernameResponse);
}

message EmptyUser {}

message GetUserRequest {
  int32 id = 1; 
}

message CreateUserRequest {
  string first_name = 1;
  string last_name = 2;   
  string username = 3;
  string email = 4;
  string password = 5;
  string photo = 6;
}

message LoginRequest {
  string username = 1;
  string password = 2;  
}

message UsernameRequest {
  string username = 1;  
}

message UsernameResponse {
  bool result = 1;
}

message User {
  int32 id = 1;
  string first_name = 2;
  string last_name = 3;
  string username = 4;
  string email = 5;
  string photo = 6;
  bool is_admin = 7;
  string api_key = 8;
}